Jenkinsfile

```diff
@@ -1,12 +1,19 @@
 pipeline {
   agent {
-    label 'linux'
+    docker {
+        image 'firedrakeproject/firedrake-env:latest'
+        label 'firedrakeproject'
+        args '-v /var/run/docker.sock:/var/run/docker.sock'
+        alwaysPull true
+        }
   }
   environment {
     PATH = "/usr/local/bin:/usr/bin:/bin"
     CC = "mpicc"
     FIREDRAKE_CI_TESTS = "1"
     PYTHONHASHSEED = "12453221"
+    DOCKER_CREDENTIALS = credentials('f52ccab9-5250-4b17-9fb6-c3f1ebdcc986')
```

Comment:
What can these credentials do? Does it matter that everyone can see them?

Reply:
Everyone can't directly see them. That's what credentials does.